KAFKA-10849: Remove useless ApiKeys#parseResponse and ApiKeys#parseRequest

dengziming opened the pull request:
*More detailed description of your change*
#7409 has removed the conversion to Struct when parsing resp, `ApiVersionsResponse.parse` will be used instead of `API_VERSIONS.parseResponse`, so the overwrite in `ApiKeys.API_VERSIONS` is useless. 

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Commit:
remove ApiKeys#parseResponse and ApiKeys#parseRequest

dengziming commented:
Hi, @chia7712 , PTAL.

chia7712 reviewed (commented):
@dengziming Thanks for your patch. Could you also remove ```ApiKeys#parseResponse``` and ```ApiKeys#parseRequest```? They are not used anymore.

dengziming commented:
@chia7712 Thank you, I also created an issue and changed the title.

chia7712 commented:
@dengziming Thanks for your updating. ```requestSchema``` and ```responseSchema``` are not used by production code also. Could you remove them?

dengziming commented:
@chia7712 I removed `responseSchema ` which was used in `ApiKeysTest.testResponseThrottleTime` to inspect all response,  I think it's similar to tests in `ApiVersionsResponseTest`, so I just also moved it.

dajac commented on clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java:
```diff
@@ -178,15 +175,7 @@
-    API_VERSIONS(18, "ApiVersions", ApiVersionsRequestData.SCHEMAS, ApiVersionsResponseData.SCHEMAS) {
-        @Override
-        public Struct parseResponse(short version, ByteBuffer buffer) {
-            // Fallback to version 0 for ApiVersions response. If a client sends an ApiVersionsRequest
-            // using a version higher than that supported by the broker, a version 0 response is sent
-            // to the client indicating UNSUPPORTED_VERSION.
-            return parseResponse(version, buffer, (short) 0);
```
@dengziming Do we maintain this fallback mechanism somewhere?

chia7712 replied:
@dajac Nice question. It is related to #7409 and the replacement is https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsResponse.java#L80

chia7712 commented:
@dengziming I just notice there is a PR (#9748) covering this one. Could you please close this PR and then be a reviewer for that PR?

dengziming commented:
@chia7712 Oh, sorry for the repeated work, I just closed this pr.
